### DISCUSSION TITLE
Need some help with rarity scheme stuff

### DISCUSSION THREAD

**Discussion by xdrushxd** [nftchef/art-engine#271] [Help + Q&A]
So i did create a art collection with "Hashlips" but they had not all the tools i needed...
Now this project could have the potential to fix my issue.
So i have 4 backgrounds and in total my collection is 100 img big.
i need the rarity of the background to look like this:

Bckground 1 = used 20 times
Bckground 2 = used 25 times
Bckground 3 = used 50 times
Bckground 4 = used 5 times

---
I can't seem to figure out how to do that and not only background but all my pictures need a specific rarity scheme.

If someone could help me i have discord.

**Comment by ghost** [comment]
You name your background images like:

Background 1#20.png
Background 2#25.png
...

The generator works with weights and the weights are indicated with #... in the filename. In your case, since you have 100 images, you can just make sure all weights for a layer add up to 100.

**Comment by BOERKEN** [comment]
> You name your background images like:
> 
> Background 1#20.png Background 2#25.png ...
> 
> The generator works with weights and the weights are indicated with #... in the filename. In your case, since you have 100 images, you can just make sure all weights for a layer add up to 100.

Hi yreuvekamp, maybe you also know if it is possible to weight the folders?
I have three different folders with different backgrounds. And i like to have them with a different rarity.
Thnaks for helping!

**Comment by ghost** [comment]
It is. You can see an example of that here: https://github.com/nftchef/art-engine#nesting-structure

**Comment by BOERKEN** [comment]
Thank you so much yreuvecamp!
goodspeed and good luck for you!
